nilearn/datasets/_utils.py

```diff
@@ -340,10 +340,13 @@ def uncompress_file(file_, delete_archive=True, verbose=1):
     """
     logger.log(f"Extracting data from {file_}...", verbose=verbose)
 
-    data_dir = os.path.dirname(file_)
+    file_ = Path(file_)
+    data_dir = file_.parent
+
     # We first try to see if it is a zip file
     try:
-        filename, ext = os.path.splitext(file_)
+        filename = data_dir / file_.stem
+        ext = file_.suffix
```

Comment:
should we replace the `ext` variable in the rest of the function with `file_.suffix`?